frontend/src/components/c-title.vue

```diff
@@ -0,0 +1,39 @@
+<template lang="pug">
+.title(v-html="markdownText", v-if="markdownText != ''")
+</template>
+
+<script lang="ts">
+import MarkdownIt from 'markdown-it';
+import { defineComponent } from 'vue';
+
+export default defineComponent({
+  data() {
+    return {
+      markdownText: '',
+    };
+  },
+  beforeMount() {
+    try {
+      fetch('title.md').then((response) => {
+        if (!response.ok) { // file not found
+          return '';
+        }
+        return response.text();
+      }).then((text) => {
+        const md = new MarkdownIt();
+        this.markdownText = md.render(text);
+      });
+    } catch (error) {
+      this.markdownText = (error as Error).toString();
```

Comment:
This is more of a Javascript "gotcha", but the catch block here doesn't catch any errors thrown by the promise chain. This is because the code doesn't wait for the promise to fulfill, and so the `fetch` function returns before the error is thrown asynchronously.

For example, if you try running this code:
```javascript
function f() {
  return Promise.reject("Error");
}

try {
  f()
    .then(() => {
      console.log("Success");
      return 1;
    })
} catch (err) {
  console.log(`Error inside try catch block: ${err}`);
}
```
You'll see that the statement inside the `catch` block never runs.

The two options we have are:
1) Using `.catch` (this makes the outer `try...catch` redundant)
```javascript
function f() {
  return Promise.reject("Error");
}

try {
  f()
    .then(() => {
      console.log("Success");
      return 1;
    })
    .catch((err) => {
      console.log(`This runs and catches the error: ${err}`)
    })
} catch (err) {
  console.log(`This code still never runs: ${err}`);
}
```

or 

2) Using `async/await`, which forces everything after await to wait for the promise to be fulfilled. So we can use the `try...catch` here:
```javascript
async function f() {
  return Promise.reject("Error");
}

try {
  await f()
  console.log("Success")
} catch (err) {
  console.log(`Now this runs: ${err}`);
}
```